libcontainer/container_linux.go

```diff
@@ -2280,3 +2309,57 @@ func requiresRootOrMappingTool(c *configs.Config) bool {
 	}
 	return !reflect.DeepEqual(c.GidMappings, gidMap)
 }
+
+// getEligibleCPU returns the first eligible CPU for CPU affinity before
+// entering in a cgroup cpuset.
+// - when there is not cpuset cores: no eligible CPU
+// - when there is not isolated cores: no eligible CPU
+// - when cpuset cores are all isolated cores: return first CPU of the cpuset
+// - when cpuset cores are mixed between housekeeping/isolated cores: no eligible CPU.
```

Comment:
@cclerget Is this rule too strict? I think we can calculate an eligible CPU. What was your idea when making this rule? Is it about algorithms’ complexity? Or something else I didn't notice?

Reply:
@lifubang Yes, originally in kubernetes context, I translated this to a non-deterministic CPU allocation and I wasn't expecting such setup to be used. But you're right, the first housekeeping CPU could be used here to handle specific CPU allocations no matter the context